store/file/ods.go

```diff
@@ -389,9 +404,14 @@ func readCol(fl io.ReaderAt, hdr *headerV0, sharesOffset, quadrantIdx, colIdx in
 		offset := sharesOffset + quadrantOffset + pos*shrLn
 
 		shr := make(share.Share, shrLn)
-		if _, err := fl.ReadAt(shr, int64(offset)); err != nil {
+		n, err := fl.ReadAt(shr, int64(offset))
+		if err != nil && !errors.Is(err, io.EOF) {
 			return nil, err
 		}
+		if n == 0 {
+			shr = share.TailPadding
+		}
+
 		shares[i] = shr
 	}
```

Comment:
Same loop allocations problem as in eds.ReadShares

Reply:
I am not sure we can solve it in this case.  ReadAt reads the whole row at once and we have to allocate it before we know its EOF. I don't think its a big deal tho as we allocate as much as we do in the regular case and with buffer pooling its gonna be even less noticable.

Reply:
The above is for readRow. For readCol we can do that, yes